frontend/src/components/common/Terminal.tsx

```diff
@@ -40,43 +54,67 @@ export default function Terminal(props: TerminalProps) {
   const [terminalContainerRef, setTerminalContainerRef] = React.useState<HTMLElement | null>(null);
   const [container, setContainer] = React.useState<string | null>(null);
   const { t } = useTranslation('resource');
+  const execRef = React.useRef<any>(null);
+  const fitAddonRef = React.useRef<any>(null);
+  const xtermRef = React.useRef<any>(null);
+  const [shells, setShells] = React.useState({
+    available: getAvailableShells(),
+    currentIdx: 0,
+    automaticallySwitch: true,
+  });
+
+  function getAvailableShells() {
+    const selector = item.spec?.nodeSelector || {};
+    const os = selector['kubernetes.io/os'] || selector['beta.kubernetes.io/os'];
+    if (os === 'linux') {
+      return ['bash', 'sh'];
+    } else if (os === 'windows') {
+      return ['powershell.exe', 'cmd.exe'];
+    }
+    return ['powershell.exe', 'bash', 'cmd.exe', 'sh'];
```

Comment:
I know this is the "last resort" statement, but I'd reorder this as `sh, bash, powershell, cmd` because ATM, Linux containers are much more widely common than Windows, meaning we'll have a better guess by using a Unix-first ordering.